packages/manager/modules/billing/src/components/services-actions/services-actions.controller.js

```diff
@@ -0,0 +1,69 @@
+import { RENEW_URL } from './service-actions.constants';
+
+export default class ServicesActionsCtrl {
+  /* @ngInject */
+  constructor(atInternet, RedirectionService, CORE_MANAGER_URLS) {
+    this.atInternet = atInternet;
+    this.RedirectionService = RedirectionService;
+    this.autorenewLink = RedirectionService.getURL('autorenew');
+
+    this.CORE_MANAGER_URLS = CORE_MANAGER_URLS;
+  }
+
+  $onInit() {
+    this.warningLink = `${this.autorenewLink}/warn-nic?nic=${this.service.contactBilling}`;
+    this.billingLink = this.RedirectionService.getURL('billing');
+    this.updateLink = `${this.autorenewLink}/update?serviceId=${this.service.serviceId}`;
+    this.cancelResiliationLink = `${this.autorenewLink}/cancel-resiliation?serviceId=${this.service.serviceId}`;
+
+    switch (this.service.serviceType) {
+      case 'EXCHANGE':
```

Comment:
Why not having all these service types in your constants file?

Reply:
`serviceType` has only few chances to change but sure I'll put them in the constants file

Reply:
it is just that I see that thoses names are used elsewhere, e.g.: https://github.com/ovh/manager/pull/2907/files#diff-ebe3111f85479ef61dfc6286d6c33db3R62 here.
But it is nitpicking